stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/textinput/MessageInputView.kt

```diff
@@ -82,6 +82,8 @@ public class MessageInputView : ConstraintLayout {
         configSendAlsoToChannelCheckbox()
     }
 
+    public var typeListener: TypeListener = EMPTY_TYPE_LISTENER
```

Comment:
I think making these listeners nullable makes for a more Android-y API, like the default things like `setOnClickListener` also accept `null` to clear the listener (instead of having to create an empty one, if you want to remove a listener).

Creating a separate setter method for these is also a good idea, again, it brings it closer to Android platform APIs (even if it's less Kotlin-y). As a bonus, for `fun interfaces` (which is not the case here, but is in a lot of other places), you can use a lambda when you're passing it into the method, but you have to use a more verbose SAM constructor if you have to set a property.

`SearchInputView` is an example of doing these things.